app.js

```diff
@@ -601,10 +603,14 @@ aws.initConfig(awsProps)
                     .use('/source', sourceHandler.handle.bind(sourceHandler))
                     .use('/api', apiHandler.handle)
                     .use('/g', oldGoogleUrlHandler)
-                    .get('/z/:id', storedStateHandler)
                     .get('/resetlayout/:id', storedStateHandlerResetLayout)
-                    .get('/clientstate/:clientstatebase64', unstoredStateHandler)
-                    .post('/shortener', storageHandler.handler.bind(storageHandler));
+                    .get('/clientstate/:clientstatebase64', unstoredStateHandler);
+
+                if (enableShort) {
```

Comment:
I don't think this setting is necessary and would require us to maintain this (+api contains functions using short link as well + /resetlayout/:id)

Reply:
Do you mean the entire PR, or just these endpoints being optionally removed? Because without giving out too much details, we need this to avoid storing files on a self-hosted instance to limit the attack surface. We could use a reverse proxy to ban these endpoints without changing compiler-explorer, but we do want to hide the option from the front-end as well

Reply:
I just meant the end-points. I understand the interest in disabling attack vectors, but an if (enableShort) around just 2 of the endpoints would require us to maintain the list everytime we add something else. (currently /resetlayout/ is also involved with shortlinks)

As an alternative I would recommend adding a storage-void.js handler and setting the property storageSolution=void to that. This way you can handle all the in/out traffic involved with shortlinks and do nothing, and if we add more endpoints to do with shortlinks we won't have to remember the if().

Reply:
Great idea, thanks!